core/config/config_helper.cpp

```diff
@@ -0,0 +1,51 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <type_traits>
+
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/config/registry.hpp>
+
+
+#include "core/config/config_helper.hpp"
+
+
+namespace gko {
+namespace config {
+
+
+type_descriptor update_type(const pnode& config, const type_descriptor& td)
+{
+    auto value_typestr = td.get_value_typestr();
+    auto index_typestr = td.get_index_typestr();
+
+    if (auto& obj = config.get("ValueType")) {
+        value_typestr = obj.get_string();
+    }
+    if (auto& obj = config.get("IndexType")) {
```

Comment:
I'm not sure we have discussed that yet, but I would prefer that the keys in the config are snake_case.
```suggestion
    if (auto& obj = config.get("value_type")) {
        value_typestr = obj.get_string();
    }
    if (auto& obj = config.get("index_type")) {
```

Reply:
I put ValueType for the type template is to match what we have in the ginkgo class

Reply:
The parameters of our factories will be specified in snake_case, so we have both spellings. We should pick only one spelling type. The snake_case for the types would not be totally inconsistent, since our classes mostly redefine the types using snake_case.

Also, the user in our current code never writes `ValueType`, but they write the parameter names, so they are more familiar with the snake_case.

Reply:
When they check how to configure the type, they will see `template <typename ValueType> Cg` -> use `ValueType: ...` to configure Cg type rather than converting the `CamelCase` to `snake_case`

Reply:
They will not look at the CG to find out how to specify the value/index types. This has to be defined at a different place, possible the parse function. There we should state how to pass in the types, and there we can decide on how to represent them.
To put it more explicitly, the user should not be asked to deduce how to define a configuration by looking at our code (except for factory parameters).